graylog2-server/src/main/java/org/graylog2/shared/bindings/providers/ParameterizedHttpClientProvider.java

```diff
@@ -0,0 +1,136 @@
+import java.security.SecureRandom;
+import java.util.concurrent.ExecutionException;
+
+@Singleton
+public class ParameterizedHttpClientProvider {
+    private final LoadingCache<Parameters, OkHttpClient> cache;
+    private final OkHttpClientProvider okHttpClientProvider;
+    private final X509TrustManager insecureTrustManager;
+    private final SSLSocketFactory insecureSocketFactory;
+    private X509TrustManager trustManager;
+    private SSLSocketFactory sslSocketFactory;
+
+    @Inject
+    public ParameterizedHttpClientProvider(OkHttpClientProvider provider) throws NoSuchAlgorithmException, KeyManagementException, KeyStoreException {
+        okHttpClientProvider = provider;
+        cache = CacheBuilder.newBuilder().build(CacheLoader.from(this::buildHttpClient));
+
+        insecureTrustManager = new TrustAllX509TrustManager();
+        SSLContext insecureSSLContext = SSLContext.getInstance("TLS");
+        insecureSSLContext.init(null, new TrustManager[]{insecureTrustManager}, new SecureRandom());
+        insecureSocketFactory = insecureSSLContext.getSocketFactory();
+
+        sslSocketFactory = SSLContext.getDefault().getSocketFactory();
+        trustManager = new DefaultX509TrustManager("ignored");
+    }
+
+    @VisibleForTesting
+    // allows testing with a custom trustManager / sslSocketFactory
+    public ParameterizedHttpClientProvider(OkHttpClientProvider provider, SSLSocketFactory sslSocketFactory, X509TrustManager trustManager) throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+        this(provider);
+        this.trustManager = trustManager;
+        this.sslSocketFactory = sslSocketFactory;
+    }
+
+    public OkHttpClient get(boolean keepAlive, boolean skipTLSVerify) {
+        try {
+            return cache.get(Parameters.fromBoolean(keepAlive, skipTLSVerify));
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private OkHttpClient buildHttpClient(Parameters parameters) {
+        final Builder builder = okHttpClientProvider.get().newBuilder();
+
+        switch (parameters) {
+            case NONE -> withDefaultSocketFactory(builder);
+            case KEEPALIVE -> withKeepAlive(builder);
+            case SKIP_TLS_VERIFY -> withSkipTLSVerification(builder);
+            case SKIP_TLS_VERIFY_AND_KEEPALIVE -> withSkipTLSVerificationAndKeepAlive(builder);
+        }
+        return builder.build();
+    }
+
+    private Builder withDefaultSocketFactory(Builder builder) {
+        builder.sslSocketFactory(sslSocketFactory, trustManager);
+        return builder;
+    }
+
+    private Builder withKeepAlive(Builder builder) {
+        builder.socketFactory(new TcpKeepAliveSocketFactory(SocketFactory.getDefault()));
+        builder.sslSocketFactory(new TcpKeepAliveSSLSocketFactory(sslSocketFactory), trustManager);
+        return builder;
+    }
+
+    private Builder withSkipTLSVerification(Builder builder) {
```

Comment:
hmm, where did you get `java:S5527` from?
IntelliJ does suggest
```
@SuppressWarnings("UnusedReturnValue")
```
and that's the only way to get rid of this warning for me

Reply:
I got this from SonarLint - I was trying to reduce the number of Sonar warnings. `S5527` isn't very descriptive unfortunately.

Reply:
> I got this from SonarLint - I was trying to reduce the number of Sonar warnings. `S5527` isn't very descriptive unfortunately.

ah, but you installed that plugin yourself? 
Does that then replace the warnings that IntelliJ usually generates?

Reply:
@patrickmann would `@SuppressWarnings("UnusedReturnValue")` also work for SonarLint? 
I can add it then.
Otherwise I don't think we should add this for a plugin that isn't used by everyone.

Reply:
Those are 2 distinct issue. Don't bother with it then.